requestmodel.py

```diff
@@ -168,15 +160,24 @@ def display_message(self, message):
 
     # this is needed for webapp2 sessions to work
     def dispatch(self):
-		# Get a session store for this request.
-		self.session_store = sessions.get_store(request=self.request)
-
-		try:
-				# Dispatch the request.
-				webapp2.RequestHandler.dispatch(self)
-		finally:
-				# Save all sessions.
-				self.session_store.save_sessions(self.response)
+        # Get a session store for this request.
+        self.session_store = sessions.get_store(request=self.request)
+        # Set locale on each request right away, to make it available to both
+        # templates and backend Python code
+        if self.request.get("locale"):
```

Comment:
Good idea to do it here. Getting the translation stuff to work is a pain if you step outside the templating system. I'll have to track down the other non-rendering places I've had to mess with it - in the email and caldav modules - maybe this cleans that up, too.